site_scons/components/__init__.py

```diff
@@ -306,19 +306,25 @@ def define_components(reqs):
     retriever = GitRepoRetriever("https://github.com/spdk/spdk.git", True)
     reqs.define('spdk',
                 retriever=retriever,
-                commands=['cd dpdk; '                            \
-                          'git fetch; '                          \
-                          'git checkout origin/spdk-19.11.6',    \
-                          './configure --prefix="$SPDK_PREFIX"'                \
-                          ' --disable-tests --without-vhost --without-crypto'  \
-                          ' --without-pmdk --without-vpp --without-rbd'        \
-                          ' --with-rdma --with-shared'                         \
-                          ' --without-iscsi-initiator --without-isal'          \
-                          ' --without-vtune', 'make $JOBS_OPT', 'make install',
-                          'cp dpdk/build/lib/* "$SPDK_PREFIX/lib"',
+                commands=['./configure --prefix="$SPDK_PREFIX"'                \
+                          ' --disable-tests --disable-unit-tests '             \
+                          ' --disable-examples --disable-apps --without-vhost '\
+                          ' --without-crypto --without-pmdk --without-rbd '    \
+                          ' --with-rdma --without-iscsi-initiator '            \
+                          ' --without-isal --without-vtune --with-shared',
+                          'make $JOBS_OPT', 'make install',
+                          'cp -r -P dpdk/build/lib/* "$SPDK_PREFIX/lib"',
+                          'mkdir -p "$SPDK_PREFIX/include/dpdk"',
+                          'cp -r -P dpdk/build/include/* '                     \
+                          '"$SPDK_PREFIX/include/dpdk"',
                           'mkdir -p "$SPDK_PREFIX/share/spdk"',
                           'cp -r include scripts "$SPDK_PREFIX/share/spdk"'],
-                libs=['rte_bus_pci'], patch_rpath=['lib'])
+                headers=['spdk/nvme.h', 'dpdk/rte_eal.h'],
+                extra_include_path=['/usr/include/dpdk',
+                                    '$SPDK_PREFIX/include/dpdk',
+                                    # debian dpdk rpm puts rte_config.h here
+                                    '/usr/include/x86_64-linux-gnu/dpdk'],
```

Comment:
But because they are providing a `pkg-config` file, they are (and can/will, without risk of breaking existing users) free to change this at any time.  Something more future-proof would be to get the list of additional includes from `pkg-config`.  Unfortunately I don't see any option to get just the paths, but a simple pipeline such as `pkg-config --cflags-only-I libdpdk | sed -e 's/-I//g'` gets you the list.  Arguably the `sed …` part could/should be done in python to avoid having to use a shell with subprocess which typically gets flagged as a security issue.

It would be good in fact to formalize this use of `pkg-config` for getting `extra_include_path` items into scons_local itself.

Reply:
I worked with @jolivier23 to try to use pkg-config but was running into other issues in scons regarding relative include directories provided within the dpdk/spdk pkg-config files this requires further investigation but shouldn't block landing the upgrade

Reply:
Yeah, I am not suggesting you use scons' `pkg-config` support as it's implemented, but merely using the `pkg-config` command to get the [list of] dir(s) to be included in `extra_include_path` rather than hard-coding a path that is subject to (unannounced), future change by the DPDK team.